pkg/controller/eks/identityproviderconfig/controller.go

```diff
@@ -0,0 +1,215 @@
+	errAddTagsFailed  = "cannot add tags to EKS identity provider config"
+)
+
+// SetupIdentityProviderConfig adds a controller that reconciles IdentityProviderConfigs.
+func SetupIdentityProviderConfig(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, poll time.Duration) error {
+	name := managed.ControllerName(v1alpha1.IdentityProviderConfigKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewController(rl),
+		}).
+		For(&v1alpha1.IdentityProviderConfig{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.IdentityProviderConfigGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newEKSClientFn: eks.NewEKSClient}),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient()), managed.NewNameAsExternalName(mgr.GetClient()), &tagger{kube: mgr.GetClient()}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithPollInterval(poll),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube           client.Client
+	newEKSClientFn func(config aws.Config) eks.Client
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.IdentityProviderConfig)
+	if !ok {
+		return nil, errors.New(errNotEKSIdentityProviderConfig)
+	}
+	cfg, err := awsclient.GetConfig(ctx, c.kube, mg, cr.Spec.ForProvider.Region)
+	if err != nil {
+		return nil, err
+	}
+	return &external{client: c.newEKSClientFn(*cfg), kube: c.kube}, nil
+}
+
+type external struct {
+	client eks.Client
+	kube   client.Client
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.IdentityProviderConfig)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotEKSIdentityProviderConfig)
+	}
+
+	rsp, err := e.client.DescribeIdentityProviderConfig(ctx, &awseks.DescribeIdentityProviderConfigInput{
+		IdentityProviderConfig: &types.IdentityProviderConfig{
+			Name: aws.String(meta.GetExternalName(cr)),
+			Type: aws.String(string(v1alpha1.OidcIdentityProviderConfigType)),
+		},
+		ClusterName: &cr.Spec.ForProvider.ClusterName})
+	if err != nil {
+		return managed.ExternalObservation{}, awsclient.Wrap(resource.Ignore(eks.IsErrorNotFound, err), errDescribeFailed)
+	}
+
+	current := cr.Spec.ForProvider.DeepCopy()
+	eks.LateInitializeIdentityProviderConfig(&cr.Spec.ForProvider, rsp.IdentityProviderConfig)
+
+	// Generate observation
+	cr.Status.AtProvider = eks.GenerateIdentityProviderConfigObservation(rsp.IdentityProviderConfig)
+
+	// Any of the statuses we don't explicitly address should be considered as
+	// the node group being unavailable.
```

Comment:
Really the nodeGroup? think during apply/delete nodeGroup is available - so Copy/Paste NitPick?

Reply:
Fixed